src/utils/window-manager/WindowManager.js

```diff
@@ -94,9 +94,15 @@ class WindowManager {
 // the default export.
 const windowManagerSingleton = new WindowManager();
 const singletonExports = {
-  registerUnloadPromptSignal: windowManagerSingleton.registerUnloadPromptSignal,
-  dangerouslyForceLocationReload: windowManagerSingleton.dangerouslyForceLocationReload,
-  dangerouslyForceLocationAssign: windowManagerSingleton.dangerouslyForceLocationAssign,
+  registerUnloadPromptSignal: (
```

Comment:
Noticed this bug when I actually started rendering it. The functions were not bound properly and did not have access to the registered prompts in the singleton.